File: Visual-Studio-Code/Python/Python-Programs/InsertionSort.py
def swap(num,one,two):
    temp = num[one]
    num[one] = num[two]
    num[two] = temp

    print(f"Swaping {num[one]} and {num[two]}")

def insertionSort(num,size):
    print("num is",type(num))
    print("size is",type(size))

    for i in range(0,size):
        k = i
        while k > 0 and num[k] > num[k-1]:
            swap(num,k,k-1)
            k -= 1


num = [6,5,4,1,8,3,2,7,9]
size = len(num)
insertionSort(num,size)

for i in range(0,size):
    print(num[i])

File: Semester-2/C/Asdos/11_Sort/InsertionSort.c
#include <stdio.h>

void swap(int *one,int *two){   
    int temp = *one;
    *one = *two;
    *two = temp;
}

int insertionSort(int *arr,int size){

    for(int i=0; i<size; i++){
        int j = i;
        while(j>0 && arr[j] <= arr[j-1]){
            swap(&arr[j],&arr[j-1]);
            j--;
        }
    }
    
    return 0;
}

int main(){

    int arr[] = {7,3,6,5,1,2,9,4,8};
    int size = sizeof(arr)/sizeof(arr[1]);
    insertionSort(arr,size);

    for(int i=0; i<size; i++){
        printf("%d ",arr[i]);
    }

    return 0;
}